packages/analytics/amplify_analytics_pinpoint/lib/src/legacy_native_data_provider/flutter_legacy_native_data_provider_io.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:io' show Platform;
+
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.android.dart';
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.ios.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+class FlutterLegacyNativeDataProvider implements LegacyNativeDataProvider {
+  /// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+  factory FlutterLegacyNativeDataProvider() {
```

Comment:
This can just be a single class for both platforms

Reply:
Not sure I understand here, there already is a single class for Android and iOS (data_provider_android.dart and data_provider_ios.dart) 

This class is what is exported via flutter_legacy_native_data_provider.

Reply:
See comment above on `data_provider.ios.dart`.